src/lesson/five/LambdaCalculator.java

```
@@ -0,0 +1,31 @@
+package lesson.five;
+
+import java.util.HashMap;
+import java.util.function.BinaryOperator;
+
+public class LambdaCalculator {
+
+    private static HashMap<Character, BinaryOperator<Double>> operations;
+
+    public LambdaCalculator() {
+        operations = addOperations();
+    }
+
+    private static HashMap<Character, BinaryOperator<Double>> addOperations() {
+        HashMap<Character, BinaryOperator<Double>> operations = new HashMap<>();
+        operations.put('+', (x, y) -> x + y);
+        operations.put('-', (x, y) -> x - y);
+        operations.put('*', (x, y) -> x * y);
+        operations.put('/', (x, y) -> x / y);
+        operations.put('#', (x, y) -> Math.pow(x, 1.0 / y));
+        operations.put('^', Math::pow);
+        operations.put('$', (x, y) -> Math.pow((x + y) / x + 117, y));
+        return operations;
+    }
+
+    public double calculate(double x, char c, double y) {
+        BinaryOperator<Double> action = operations.get(c);
```

What if you don't have such an operation? In this case better to throw a custom exception.